### ISSUE TITLE
Updating the look of the Component Browser to the new design

### ISSUE THREAD

**Issue by wdanilo** [enso-org/enso#4749]
This task is automatically imported from the old Task Issue Board and it was originally created by Adam Obuchowicz.
Original issue is [here](https://www.pivotaltracker.com/story/show/183313944).
<hr />

## Why

**As** Enso IDE user
**I want** to have CB in style fitting the rest of the GUI
**So that** The GUI is nice looking.

The CB should look like on [the first screen in the design doc](https://user-images.githubusercontent.com/1623053/183441941-95205e94-7af1-4f12-abbb-823206554117.png) **except blurred background**

## Acceptance Criteria

The screenshot from the new IDE and the screen from the design docs looks the same.

## Blockers:
[#182610422](https://www.pivotaltracker.com/story/show/#182610422) resolved

## Comments:
**Ilya Bogdanov** reports a new **STANDUP** for yesterday (2022-10-18):

**Progress:** started the implementation. Adjusted a few parameters in the style sheet, changed section navigator to use grid view instead of list view. It should be finished by 2022-10-24. (Enso Bot - Oct 19, 2022)
<hr />
**Ilya Bogdanov** reports a new **STANDUP** for today (2022-10-19):

**Progress:** adjusted the section navigator settings to match the design, implemented support of different colors for section buttons, and added colors animation. It should be finished by 2022-10-24.

**Next Day:** Next day I will be working on the [same task](https://www.pivotaltracker.com/story/show/183313944). continue working on adjusting the styles. Breadcrumbs are on the way (Enso Bot - Oct 19, 2022)
<hr />
**Ilya Bogdanov** reports a new **STANDUP** for yesterday (2022-10-20):

**Progress:** Adjusted the look of the icons in the section navigator (they changed a bit in a new design) and made multiple adjustments in the breadcrumbs. I also tweaked the colors of the whole browser. (not finished yet) It should be finished by 2022-10-24.

**Next Day:** Next day I will be working on the [same task](https://www.pivotaltracker.com/story/show/183313944). continue updating the design (Enso Bot - Oct 21, 2022)
<hr />
**Ilya Bogdanov** reports a new **STANDUP** for the last Friday (2022-10-21):

**Progress:** Tried changing a font, but faced an issue with M1Plus that makes its use impossible. Started reworking icons of the components. It should be finished by 2022-10-24.

**Next Day:** Next day I will be working on the [same task](https://www.pivotaltracker.com/story/show/183313944). continue updating the design (Enso Bot - Oct 24, 2022)
<hr />
**Ilya Bogdanov** reports a new **🔴 DELAY** for yesterday (2022-10-24):

**Summary:** There is **2 days** delay in implementation of the [Updating the look of the Component Browser to the new design (#183313944) task](https://www.pivotaltracker.com/story/show/183313944).
It **will cause 0 days delay** for the delivery of this weekly plan.

**Delay Cause:** changing font is blocked by renderer bug, and also updating icons was not estimated initially (Enso Bot - Oct 25, 2022)
<hr />
**Ilya Bogdanov** reports a new **STANDUP** for yesterday (2022-10-24):

**Progress:** Updating more icons to the new look. Trying to adjust the colors of the component groups. I learned that we must change our implementation slightly, especially the group dimming and icon colors when the entry is highlighted. I consulted with Wojciech about colors and transparent background. Attended weekly team demo and 1-1 meeting. It should be finished by 2022-10-26.

**Next Day:** Next day I will be working on the [same task](https://www.pivotaltracker.com/story/show/183313944). continue updating the design, possibly preparing a WIP PR with changes (Enso Bot - Oct 25, 2022)
<hr />
**Ilya Bogdanov** reports a new **STANDUP** for yesterday (2022-10-25):

**Progress:** updated the rest of the icons, fixed the look of the dimmed component groups. Started reworking the colors of the highlighted entries. It should be finished by 2022-10-26. (Enso Bot - Oct 26, 2022)
<hr />
**Ilya Bogdanov** reports a new **🔴 DELAY** for yesterday (2022-10-26):

**Summary:** There is **2 days** delay in implementation of the [Updating the look of the Component Browser to the new design (#183313944) task](https://www.pivotaltracker.com/story/show/183313944).
It **will cause 0 days delay** for the delivery of this weekly plan.

**Delay Cause:** refactorings for proper color management take time (Enso Bot - Oct 27, 2022)
<hr />
**Ilya Bogdanov** reports a new **STANDUP** for yesterday (2022-10-26):

**Progress:** Tried using transparent background one more time and succeeded. It turns out I messed up rgb values and oversaturated the color during previous attempt. Implemented a PoC solution for handling both arbitrary colors and intensities through a single stylesheet entry. It will allow us to implement a proper design. Adjusting colors, fighting with Lcha colorspace. There is no good way to calculate Lcha parameters to use in our APIs, you need to use Rgba -> Lcha conversion instead. It should be finished by 2022-10-28. (Enso Bot - Oct 27, 2022)
<hr />
**Ilya Bogdanov** reports a new **STANDUP** for today (2022-10-27):

**Progress:** Refactoring and polishing the code, preparing a PR, using new intensity/color API for every color parameter in component groups. It should be finished by 2022-10-28.

**Next Day:** Next day I will be working on the [same task](https://www.pivotaltracker.com/story/show/183313944). finishing the PR with the first part of the design update (Enso Bot - Oct 27, 2022)
<hr />
**Ilya Bogdanov** reports a new **🔴 DELAY** for the last Friday (2022-10-28):

**Summary:** There is **5 days** delay in implementation of the [Updating the look of the Component Browser to the new design (#183313944) task](https://www.pivotaltracker.com/story/show/183313944).
It **will cause 0 days delay** for the delivery of this weekly plan.

2 days are weekends, and 1 day is a day off

**Delay Cause:** the correct colors are unknown, and the correct font is not supported. Also preparing a PR takes a lot of time due to important refactorings. (Enso Bot - Oct 31, 2022)
<hr />
**Ilya Bogdanov** reports a new **STANDUP** for the last Friday (2022-10-28):

**Progress:** Refactorings, writing documentation, solving minor issues with styles. It should be finished by 2022-11-02. (Enso Bot - Oct 31, 2022)
<hr />
**Ilya Bogdanov** reports a new **STANDUP** for the provided date (2022-10-31):

**Progress:** Merged with the most recent develop, integrated a popup for marketplace icon. After merge I noticed issues with layers order for section navigator icons. Fixing it required major refactoring with moving some code to a dedicated crate. Attended a weekly demo. It should be finished by 2022-11-02. (Enso Bot - Nov 4, 2022)
<hr />
**Ilya Bogdanov** reports a new **🔴 DELAY** for the last Friday (2022-11-04):

**Summary:** There is **7 days** delay in implementation of the [Updating the look of the Component Browser to the new design (#183313944) task](https://www.pivotaltracker.com/story/show/183313944).
It **will cause 0 days delay** for the delivery of this weekly plan.

2 days are weekends, 2 days were spent working on other tasks, so the actual delay is closer to 3 days

**Delay Cause:** additional work is needed to address the concerns spotted during the review. (Enso Bot - Nov 7, 2022)
<hr />
**Ilya Bogdanov** reports a new **STANDUP** for the last Friday (2022-11-04):

**Progress:** removed second color from component browser icons, changed the algorithm of color calculations of the component groups. QA testing of #183390749. Spotted performance issues in the grid_view debug scene, checked a few commits from develop to verify a regression. It should be finished by 2022-11-09. (Enso Bot - Nov 7, 2022)
<hr />
**Ilya Bogdanov** reports a new **STANDUP** for yesterday (2022-11-07):

**Progress:** Tested #183390749 one more time, accepted as the component browser works well. Refactoring and polishing the code. Rebased on top of the current develop, integrated latest changes by WD. Found two issues: one related to navigator buttons, and another one related to selection colors for dimmed component groups. Attended weekly demo. It should be finished by 2022-11-09. (Enso Bot - Nov 8, 2022)
<hr />
**Ilya Bogdanov** reports a new **STANDUP** for yesterday (2022-11-08):

**Progress:** Fixed both remaining issues, updated code docs, updated PR description, and answered all review comments. Started working on updating font and text styling in breadcrumbs and component entries. It should be finished by 2022-11-09.

**Next Day:** Next day I will be working on the [same task](https://www.pivotaltracker.com/story/show/183313944). finishing updating font, returning to the new US about skip and freeze macros (Enso Bot - Nov 8, 2022)
<hr />
**Ilya Bogdanov** reports a new **🔴 DELAY** for yesterday (2022-11-14):

**Summary:** There is **7 days** delay in implementation of the [Updating the look of the Component Browser to the new design (#183313944) task](https://www.pivotaltracker.com/story/show/183313944).
It **will cause 0 days delay** for the delivery of this weekly plan.

2 days are weekends, and two days were spent working on another task (skip and freeze macros)

**Delay Cause:** review + icon shapes ordering issue (Enso Bot - Nov 15, 2022)
<hr />
**Ilya Bogdanov** reports a new **STANDUP** for yesterday (2022-11-14):

**Progress:** created WIP PR for component browser font updates, and published changes to the first design update PR. One of the review comments is not addressed, because I faced weird shape ordering issue with component browser icons. Debugged the issue, didn't reproduce it on a simpler demo scene but found out that it is related to the scroll area layers infrastructure. Attended a weekly demo. It should be finished by 2022-11-16. (Enso Bot - Nov 15, 2022)
<hr />
**Ilya Bogdanov** reports a new **STANDUP** for yesterday (2022-11-16):

**Progress:** polishing the code and minor changes. Found out that the section navigator was accidentally broken, debugged, and fixed the issue. Created pivotal tasks for future work on component browser design-related updates. It should be finished by 2022-11-16.

**Next Day:** Next day I will be working on the [#183413872 task](https://www.pivotaltracker.com/story/show/183413872). working on skip and freeze macros (Enso Bot - Nov 17, 2022)
<hr />
**Ilya Bogdanov** reports a new **🔴 DELAY** for yesterday (2022-11-29):

**Summary:** There is **15 days** delay in implementation of the [Updating the look of the Component Browser to the new design (#183313944) task](https://www.pivotaltracker.com/story/show/183313944).
It **will cause 0 days delay** for the delivery of this weekly plan.

**Delay Cause:** the delay is significant, but the time was mostly spent working on different tasks while waiting for review. Also, the initial task was seriously underestimated because of the needed refactorings. (Enso Bot - Nov 30, 2022)
<hr />
**Ilya Bogdanov** reports a new **STANDUP** for yesterday (2022-11-29):

**Progress:** fixed remaining minor issues with the design and opened a new PR It should be finished by 2022-12-01. (Enso Bot - Nov 30, 2022)
<hr />
**Ilya Bogdanov** reports a new **🔴 DELAY** for the last Friday (2022-12-02):

**Summary:** There is **1 day** delay in implementation of the [Updating the look of the Component Browser to the new design (#183313944) task](https://www.pivotaltracker.com/story/show/183313944).
It **will cause 0 days delay** for the delivery of this weekly plan.

**Delay Cause:** working on SKIP and FREEZE macros (Enso Bot - Dec 5, 2022)
<hr />
**Ilya Bogdanov** reports a new **STANDUP** for the last Friday (2022-12-02):

**Progress:** This is a standup for 2022-12-01, because the bot was down. Fixing minor issues with the latest design iteration. Adjusted the position of group headers, section navigator separator, and section navigator buttons. It should be finished by 2022-12-02. (Enso Bot - Dec 5, 2022)
<hr />
